influxdb3_write/src/catalog.rs

```diff
@@ -319,6 +319,19 @@ impl TableDefinition {
         self.schema = schema;
     }
 
+    pub(crate) fn index_columns(&self) -> Vec<String> {
```

Comment:
I'm not sure that we'll want to continue to default to having all tag columns be indexed. We'll need a new API & CLI to set the columns explicitly that a user wants indexed. We may find that by default we don't want to index anything.